BenchmarkSample/ui/src/main/java/com/example/benchmark/ui/MainActivity.kt

```diff
@@ -0,0 +1,60 @@
+package com.example.benchmark.ui
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+
+class MainActivity : AppCompatActivity() {
+    val adapter = SampleAdapter()
+    lateinit var recyclerView: RecyclerView
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_main)
+
+        recyclerView = findViewById(R.id.recyclerview)
+        recyclerView.adapter = adapter
+    }
+}
+
+class Holder(root: View) : RecyclerView.ViewHolder(root) {
+    private val textView: TextView = root.findViewById(R.id.label)
+
+    fun bind(string: String) {
+        textView.text = string
+    }
+
+    companion object {
+        fun create(parent: ViewGroup): Holder {
+            val inflater = LayoutInflater.from(parent.context)
+            return Holder(inflater.inflate(R.layout.item_view, parent, false))
+        }
+    }
+}
+
+class SampleAdapter : ListAdapter<String, Holder>(StringDiffCallback) {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
+        return Holder.create(parent)
+    }
+
+    override fun onBindViewHolder(holder: Holder, position: Int) {
+        System.out.println("binding viewholder $position");
+        holder.bind(getItem(position))
+    }
+}
+
+private object StringDiffCallback : DiffUtil.ItemCallback<String>() {
+    override fun areItemsTheSame(oldItem: String, newItem: String): Boolean =
```

Comment:
I think `areItemsTheSame` is supposed to return referential equality, `===`

Reply:
Since content isn't changing, it doesn't matter, but you generally don't ever want reference equality in areItemsTheSame, you usually want an ID compare.

Reference equality can mess you up if e.g. you've reloaded the items from the DB, so they'll be different instances.